File: app/src/main/java/com/gongbo/fss/demo/MainActivity.java
package com.gongbo.fss.demo;

import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.ListView;

import com.gongbo.fss.adapter.listview.CommonAdapter;
import com.gongbo.fss.base.BaseFssActivity;
import com.gongbo.fss.bind.annotation.BindActivity;
import com.gongbo.fss.bind.annotation.BindView;
import com.gongbo.fss.router.annotation.Route;

import java.util.Arrays;
import java.util.List;

import static com.gongbo.fss.router.FssRouteApi.DefaultRouteApi;

@Route
@BindActivity(layout = R.layout.activity_list_view)
public class MainActivity extends BaseFssActivity implements AdapterView.OnItemClickListener {

    @BindView(id = R.id.img_back)
    private ImageView imgBack;

    @BindView(id = R.id.list_view)
    private ListView listView;

    private List<String> datas = Arrays.asList(
            "RunPriorityTestActivity",
            "BindTestActivity",
            "RouteTestActivity",
            "AdapterActivity"
    );

    @Override
    protected void initView() {
        super.initView();
        imgBack.setVisibility(View.GONE);
        listView.setAdapter(new CommonAdapter<>(this, datas, R.layout.layout_list_item,
                (holder, s, position) -> holder.setText(R.id.tv_text, s)));
        listView.setOnItemClickListener(this);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        switch (position) {
            case 0:
                DefaultRouteApi.navigateToRunPriorityTestActivity(this);
                break;
            case 1:
                DefaultRouteApi.navigateToBindTestActivity(this);
                break;
            case 2:
                DefaultRouteApi.navigateToRouteMainActivity(this);
                break;
            case 3:
                DefaultRouteApi.navigateToAdapterActivity(this);
                break;
            default:
                throw new IllegalStateException("Unexpected value: " + position);
        }
    }
}

File: app/src/main/java/com/gongbo/fss/demo/MainActivity.kt
package com.gongbo.fss.demo

import android.view.View
import android.widget.AdapterView
import android.widget.ImageView
import android.widget.ListView

import com.gongbo.fss.adapter.listview.CommonAdapter
import com.gongbo.fss.base.BaseFssActivity
import com.gongbo.fss.bind.annotation.BindActivity
import com.gongbo.fss.bind.annotation.BindView
import com.gongbo.fss.router.annotation.Route

import java.util.Arrays

import com.gongbo.fss.router.FssRouteApi.DefaultRouteApi

@Route
@BindActivity(layout = R.layout.activity_list_view)
class MainActivity : BaseFssActivity(), AdapterView.OnItemClickListener {

    @BindView(id = R.id.img_back)
    private val imgBack: ImageView? = null

    @BindView(id = R.id.list_view)
    private val listView: ListView? = null

    private val datas = Arrays.asList(
            "RunPriorityTestActivity",
            "BindTestActivity",
            "RouteTestActivity",
            "AdapterActivity"
    )

    override fun initView() {
        super.initView()
        imgBack!!.visibility = View.GONE
        listView!!.adapter = CommonAdapter(this, datas, R.layout.layout_list_item
        ) { holder, s, _ -> holder.setText(R.id.tv_text, s) }
        listView.onItemClickListener = this
    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        when (position) {
            0 -> DefaultRouteApi.navigateToRunPriorityTestActivity(this)
            1 -> DefaultRouteApi.navigateToBindTestActivity(this)
            2 -> DefaultRouteApi.navigateToRouteMainActivity(this)
            3 -> DefaultRouteApi.navigateToAdapterActivity(this)
            else -> throw IllegalStateException("Unexpected value: $position")
        }
    }
}
